patches/server/0201-Implement-EntityTeleportEndGatewayEvent.patch

```diff
@@ -5,27 +5,27 @@ Subject: [PATCH] Implement EntityTeleportEndGatewayEvent
 
 
 diff --git a/src/main/java/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
-index 9f685b4fe6d57d921e9e7a4d817e0913b863c0ad..11f9b40d0cfb77d69ecd2c7ef71dc409d1d79eba 100644
+index 020314dded2c6305bd5e9013b108bc74be25ccc1..85914124014b4e6f0a561cf560918af68682b6f5 100644
 --- a/src/main/java/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
 +++ b/src/main/java/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
-@@ -225,9 +225,20 @@ public class TheEndGatewayBlockEntity extends TheEndPortalBlockEntity {
+@@ -224,9 +224,20 @@ public class TheEndGatewayBlockEntity extends TheEndPortalBlockEntity {
  
                  }
                  // CraftBukkit end
-+                // Paper start - EntityTeleportEndGatewayEvent - replicated from above
-+                org.bukkit.craftbukkit.entity.CraftEntity bukkitEntity = entity.getBukkitEntity();
-+                org.bukkit.Location location = new org.bukkit.Location(world.getWorld(), (double) blockposition1.getX() + 0.5D, (double) blockposition1.getY() + 0.5D, (double) blockposition1.getZ() + 0.5D);
-+                location.setPitch(bukkitEntity.getLocation().getPitch());
-+                location.setYaw(bukkitEntity.getLocation().getYaw());
++                // Paper start - EntityTeleportEndGatewayEvent
++                org.bukkit.Location location = new org.bukkit.Location(world.getWorld(), blockposition1.getX() + 0.5D, blockposition1.getY(), blockposition1.getZ() + 0.5D);
++                location.setPitch(entity1.getXRot());
++                location.setYaw(entity1.getBukkitYaw());
 +
++                org.bukkit.entity.Entity bukkitEntity = entity1.getBukkitEntity();
 +                com.destroystokyo.paper.event.entity.EntityTeleportEndGatewayEvent event = new com.destroystokyo.paper.event.entity.EntityTeleportEndGatewayEvent(bukkitEntity, bukkitEntity.getLocation(), location, new org.bukkit.craftbukkit.block.CraftEndGateway(world.getWorld(), blockEntity));
 +                if (!event.callEvent()) {
 +                    return;
 +                }
 +                // Paper end
  
                  entity1.setPortalCooldown();
--                entity1.teleportToWithTicket((double) blockposition1.getX() + 0.5D, (double) blockposition1.getY(), (double) blockposition1.getZ() + 0.5D);
+-                entity1.teleportToWithTicket((double) blockposition1.getX() + 0.5D, (double) blockposition1.getY(), (double) blockposition1.getZ() + 0.5D); // Paper - diff on change
 +                entity1.teleportToWithTicket(event.getTo().getX(), event.getTo().getY(), event.getTo().getZ()); // Paper
```

Comment:
Just a comment on something I noticed later, I wonder if it'd be a nicer diff to reassign the `blockposition1` local var with the location from the event, that way this diff could be avoided. Not required for merge, but it could better to avoid the replacement diff (so we can keep the diff on change comment actually valid)

Reply:
Yes that would be convenient but then plugin can't change the offset anymore. The possible way would be to compare if the destination location has changed and keep that line if it doesn't change